lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -309,10 +312,58 @@ void StagesSubject::stageChanged(
 
         if (!InTransform3dChange::inTransform3dChange()) {
             // Is the change a Transform3d change?
+            const UsdPrim prim = stage->GetPrimAtPath(changedPath.GetPrimPath());
             const TfToken nameToken = changedPath.GetNameToken();
             if (nameToken == UsdGeomTokens->xformOpOrder || UsdGeomXformOp::IsXformOp(nameToken)) {
                 Ufe::Transform3d::notify(ufePath);
                 UFE_V2(sendValueChangedFallback = false;)
+            } else if (prim && prim.IsA<UsdGeomPointInstancer>()) {
+                // If the prim at the changed path is a PointInstancer, check
+                // whether the modified path is one of the attributes authored
+                // by point instance manipulation.
+                if (nameToken == UsdGeomTokens->orientations
+                    || nameToken == UsdGeomTokens->positions
+                    || nameToken == UsdGeomTokens->scales) {
+                    // This USD change represents a Transform3d change to a
+                    // PointInstancer prim.
+                    // Unfortunately though, there is no way for us to know
```

Comment:
Ouch.  We have to find a way to fix this.  Certainly the Transform3d interface knows which point instance index it's dealing with.  There has to be a way to leverage that.

Reply:
I think the most recent changes largely address this. The point instance manipulation commands now block this handling from running using the existing UfeNotifGuard mechanism, and they issue the Transform3d notifications themselves during `undo()` and `redo()`:
https://github.com/Autodesk/maya-usd/pull/1241/commits/b6575883e4b9d76786970459eebc114ec4d850f6#diff-26f9c9bdd71eb3cbf8eb499ab811fe928d671f8f6038293209b8dd08fce76dc2R73

This all instances blast still needs to remain though for any "external" authoring that happens in USD.